### PR TITLE
eliminate dead store operations that caused scan-build failures

### PR DESCRIPTION
**Pull request by rcsanchez97**
Evergreen patch build: https://spruce.mongodb.com/version/623cd28b61837d50964ea4e1/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&taskName=scan-build (it is a complete patch build, but the link has it filtered down to only scan-build tasks)

@vector-of-bool I requested your review as well since this required tweaking code that was committed by you in the last few months.

### CODE REVIEW

File: src/libmongoc/tests/test-mongoc-topology.c [hunk 1]
```diff
@@ -1747,7 +1747,7 @@ test_compatible_null_error_pointer (void)
    mock_server_run (server);
    client =
       test_framework_client_new_from_uri (mock_server_get_uri (server), NULL);
-   td = mc_tpld_unsafe_get_const (client->topology);
+   mc_tpld_unsafe_get_const (client->topology);
```
Comment (vector-of-bool) [inline src/libmongoc/tests/test-mongoc-topology.c:1750]: This line can be deleted outright.

**Review by vector-of-bool — APPROVED** [review]
LGTM, but one note.

**Review by kevinAlbs — APPROVED** [review]

File: src/libmongoc/tests/test-mongoc-topology.c (hunk 1 above)
Comment (rcsanchez97) [reply src/libmongoc/tests/test-mongoc-topology.c]: @vector-of-bool Thanks for pointing that out.  I assumed that the call to
`mc_tpld_unsafe_get_const` produced a side-effect that might not be okay
to eliminate.  I just looked at the implementation and I can see that it
does not have any side-effects.

**Comment by rcsanchez97** [comment]
Thanks for pointing that out.  I assumed that the call to
`mc_tpld_unsafe_get_const` produced a side-effect that might not be okay
to eliminate.  I just looked at the implementation and I can see that it
does not have any side-effects.

**Pull request merged**